### ISSUE TITLE
Error in rule root_tree

### ISSUE THREAD

**Issue by jmtsuji** [metagenome-atlas/atlas#233]
I'm using ATLAS `2.1.4` on a Linux server (not a cluster) running Ubuntu 18.04.

After running the workaround stated in issue #231, I get a separate `ImportError` in `rule root_tree`:

```
Traceback (most recent call last):
  File "/Analysis/jmtsuji/Cfx3/.snakemake/scripts/tmparcfn_w4.tree.py", line 11, in <module>
    from . import parsers_checkm
ImportError: cannot import name 'parsers_checkm'
[Fri Sep 20 11:00:18 2019]
Error in rule root_tree:
    jobid: 3
    output: genomes/tree/tree.nwk
    conda-env: /Data/reference_databases/atlas/2.1.x/conda_envs/5bf853c6
    
    RuleException:
CalledProcessError in line 31 of /home/jmtsuji/miniconda3/envs/atlas_2.1.4/lib/python3.6/site-packages/atlas/rules/tree.smk:
Command 'source /home/jmtsuji/miniconda3/bin/activate '/Data/reference_databases/atlas/2.1.x/conda_envs/5bf853c6'; set -euo pipefail;  python /Analysis/jmtsuji/Cfx3/.snakemake/scripts/tmparcfn_w4.tree.py' returned non-zero exit status 1.
  File "/home/jmtsuji/miniconda3/envs/atlas_2.1.4/lib/python3.6/site-packages/atlas/rules/tree.smk", line 31, in __rule_root_tree
  File "/home/jmtsuji/miniconda3/envs/atlas_2.1.4/lib/python3.6/concurrent/futures/thread.py", line 56, in run
Job failed, going on with independent jobs.
```

For some reason, the `tree.py` seems unable to load `parsers_checkm` in:
```
from . import parsers_checkm
```

Any insights here? Thanks.

**Comment by tianlei22** [comment]
I met the same error.

**Comment by jmtsuji** [comment]
@SilasK Thanks for working on this. Did you notice my comment at https://github.com/metagenome-atlas/atlas/pull/234 ? I commented after the PR was merged, so it's possible you missed it.

**Comment by jmtsuji** [comment]
@SilasK I could make a PR for this to the master branch if helpful.

**Comment by SilasK** [comment]
e4c0069

Solved this, didn't it?

**Comment by jmtsuji** [comment]
@SilasK  One additional bug at the end of the script:

```
phyla= parsers_checkm.load_checkm_tax(snakemake.input.taxonomy).phylum
```

Should be:
```
phyla= load_checkm_tax(snakemake.input.taxonomy).phylum
```
(because `load_checkm_tax` is now a function within the script itself)

**Comment by SilasK** [comment]
Thank you.

**Comment by jmtsuji** [comment]
@SilasK Thanks -- looks good.

**Comment by Sofie8** [comment]
Hello Jackson, Silas,

I was using atlas 2.14, but I installed the dev version, 2.1.6 with the suggested fixes.

I included your suggestion Jackson:
phyla= parsers_checkm.load_checkm_tax(snakemake.input.taxonomy).phylum
to:
phyla= load_checkm_tax(snakemake.input.taxonomy).phylum

However, I receive another error now:

[Thu Oct 10 03:02:02 2019]
localrule root_tree:
    input: genomes/checkm/storage/tree/fasttree.nwk, genomes/checkm/taxonomy.tsv
    output: genomes/tree/tree.nwk
    jobid: 2

Activating conda environment: /ddn1/vol1/site_scratch/leuven/314/vsc31426/db/atlas2/conda_envs/f68a61bc
Traceback (most recent call last):
  File "/ddn1/vol1/site_scratch/leuven/314/vsc31426/Shotgun/Francisco/.snakemake/scripts/tmp31y2a2av.tree.py", line 76, in <module>
    phyla= load_checkm_tax(snakemake.input.taxonomy).phylum
  File "/ddn1/vol1/site_scratch/leuven/314/vsc31426/Shotgun/Francisco/.snakemake/scripts/tmp31y2a2av.tree.py", line 28, in load_checkm_tax
    checkmTax.columns=['kindom','phylum','class','order','family','genus','species']
  File "/ddn1/vol1/site_scratch/leuven/314/vsc31426/db/atlas2/conda_envs/f68a61bc/lib/python3.6/site-packages/pandas/core/generic.py", line 3094, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/src/properties.pyx", line 65, in pandas._libs.lib.AxisProperty.__set__ (pandas/_libs/lib.c:45255)
  File "/ddn1/vol1/site_scratch/leuven/314/vsc31426/db/atlas2/conda_envs/f68a61bc/lib/python3.6/site-packages/pandas/core/generic.py", line 473, in _set_axis
    self._data.set_axis(axis, labels)
  File "/ddn1/vol1/site_scratch/leuven/314/vsc31426/db/atlas2/conda_envs/f68a61bc/lib/python3.6/site-packages/pandas/core/internals.py", line 2836, in set_axis
    (old_len, new_len))
ValueError: Length mismatch: Expected axis has 5 elements, new values have 7 elements

Any solutions for this?
Thank you so much.

**Comment by SilasK** [comment]
You have pretty unknown genomes :-).
None of them had a species annotation.
I have to check this.

You can change to the branch 'gtdb' see PR. It uses the GTDB for Taxonomy
annotation and builds a tree.

**Comment by Sofie8** [comment]
Switching to the GTDB branch indeed worked! Thanks!

On a side note, with the install of conda_env for gtdb and also das_tool, my cluster HPC system asks each time for, and exits:
libstdcxx-ng,  libgcc-ng

NoPackagesFoundError: Dependencies missing in current linux-64 channels:
  - gtdbtk 0.3.* -> fastani -> libstdcxx-ng >=4.9
  - gtdbtk 0.3.* -> fastani -> libgcc-ng >=7.3.0

How can I specify it in the yaml file, that it first downloads this libstdcxx and libgcc (anaconda) and then the gtdb or das_tool packages?

**Comment by SilasK** [comment]
Hey @Sofie8 cool to hear from you.

**Comment by SilasK** [comment]
@sofie8 If [this](https://github.com/conda-forge/conda-forge.github.io/issues/763) is the explication we should all uptdate to conda 4.6

**Comment by SilasK** [comment]
Fixed by #257
